### PR TITLE
STONEBLD-845 use a mutex rather than a sync map

### PR DESCRIPTION
**Pull request by stuartwdouglas**
If there are multiple notifactions for the same DB in quick success the map access could race, resulting in an inconsistent state.

### CODE REVIEW

File: openshift-with-appstudio-test/e2e/periodic_test.go [hunk 1]
```diff
@@ -144,25 +167,34 @@ func TestServiceRegistry(t *testing.T) {
 				if !ok {
 					continue
 				}
-				s, k := state.Load(db.Name)
+				stateMutex.Lock()
```
Comment (stuartwdouglas) [inline openshift-with-appstudio-test/e2e/periodic_test.go:170]: @gabemontero we occasionally see the tests fail, even though all builds have completed successfully, with a the building count being 1 and the completed count not being updated. 

My theory is that if you get to notifications for the same DependencyBuild in quick success this state map might be racey.

Although as I am writing this I have realized that this won't help if there is a chance notification can be processed out of order, so this PR probably won't help.

Basically if I a `DependencyBuildStateBuilding` and a `DependencyBuildStateComplete` notification triggered basically at the same time, then if the `DependencyBuildStateComplete` notification is processed before the `DependencyBuildStateBuilding` notification then the completion and building count will not match reality. 

I need to think about this some more.

Comment (stuartwdouglas) [reply openshift-with-appstudio-test/e2e/periodic_test.go:170]: That would work, but would mean that when the flake happens it would still wait for timeout before reporting success. Maybe the best solution is to just have a periodic check on the actual state of everything outstanding to make sure there is no flakes, which is a bit yuck but probably the best way to handle it.

Comment (stuartwdouglas) [reply openshift-with-appstudio-test/e2e/periodic_test.go:170]: I have gone for adding a 'freshness check' in the sync block, so we only actually process the state transition if the object is current. 

This should stop state transitions from being processed out of order.

File: openshift-with-appstudio-test/e2e/periodic_test.go [hunk 2]
```diff
@@ -144,25 +167,53 @@ func TestServiceRegistry(t *testing.T) {
...
+				//net result will be correct as the state changes themselves will be processed in the correct order
+				s, k := stateMap[db.Name]
+				current, err := jvmClient.JvmbuildserviceV1alpha1().DependencyBuilds(ta.ns).Get(ctx, db.Name, metav1.GetOptions{})
+				if err != nil {
+					ta.Logf("Failed to get latest DependencyBuild version" + err.Error())
+					stateMutex.Unlock()
+					break
+				}
+				if current.ResourceVersion != db.ResourceVersion {
+					ta.Logf(fmt.Sprintf("Skipping watch event as DependencyBuild %s was not the most recent version", db.Name))
```
Comment (stuartwdouglas) [inline openshift-with-appstudio-test/e2e/periodic_test.go:186]: Looking at how many times this message is generated I think this is clearly not just theoretical.
